packages/zeplin-extension-style-kit/declarations/margin.js

```diff
@@ -0,0 +1,28 @@
+import { STYLE_PROPS } from "../constants";
+import Area from "../values/area";
+
+const useRemUnitForMeasurement = ({ useForMeasurements }) => useForMeasurements;
+
+class Margin {
+    static get Zero() {
+        return new Margin(new Area({ top: 0, right: 0, bottom: 0, left: 0 }, { useRemUnitForMeasurement }));
```

Comment:
```
  Margin({ top: Length, right: Length, bottom: Length, left: Length }, { useRemUnitForMeasurement }));
  Padding({ top: Length, right: Length, bottom: Length, left: Length }, { useRemUnitForMeasurement }));
```

`Area` might be a little misleading here since these properties do not describe an area, they are the values for 4 direction (top, right, bottom, left). 

So, we can get rid of `Area` abstraction and pass these parameters directly to margin/padding declarations.

Reply:
I think it's better to go with the `Area`. It gives us a pattern like separation of concerns. It also fits with other classes. the constructor of declaration gets a value instance, it uses that instance and provides an abstraction between code snippet generation and values. The name of `Area` comes from MDN actually https://developer.mozilla.org/en-US/docs/Web/CSS/margin https://developer.mozilla.org/en-US/docs/Web/CSS/padding. They mention the `margin area` and `padding area`. Alternatively, we can change the name `Area`. But I couldn't find it. 😅

Reply:
Hmm, I see. My understanding is that margin area is combination of four margin segments, which are `margin-top * width`, `margin-left * height`, and so on. On the other hand, what we call `Area` does not represent the same thing, it denotes the length/thickness of margin on four sides. That's why I find this misleading/confusing.

> It gives us a pattern like separation of concerns.

I agree, it would be address this in a single place, but we can still take advantage of this through an internal type to represent this 4-valued construct. So, `Margin` would use it internally to determine equality and to generate value string etc. But, this doesn't need to be visible to outside as an additional value type of a type for a parameter in the constructor of `Margin`.
 
My point is that we're free to keep this as an implementation detail. If we think this is a right abstraction to add to the API surface, we can expose it as a value with a proper name otherwise we can keep it internal. In case it's an internal detail, it shouldn't determine the API we expose in `zeplin-extension-style-kit` package. 

Looking at the formal syntax for margin, for example, its value is specified as [this](https://developer.mozilla.org/en-US/docs/Web/CSS/margin):
```
[ <length> | <percentage> | auto ]{1,4}
```

We tend to stick to the CSS specification to determine if we should have a standalone concept for something or not. Adding a custom value would change this, that's why I a little resistant to do such thing. 😅 

It might be good to get more opinions on this. @merih WDYT?

----
Reminder:
> It also fits with other classes. the constructor of declaration gets a value instance, it uses that instance and provides an abstraction between code snippet generation and values. 

This is not a strict rule or there's no structural assumptions made based on this. See `Border` below for example. So, this shouldn't be a constraint while defining a declaration. 
https://github.com/zeplin/stylesheet-extensions/blob/bdac86c534e9ca22e3ce27b125f29ba36f9cbb90/packages/zeplin-extension-style-kit/declarations/border.d.ts#L6-L14